pinot-core/src/main/java/org/apache/pinot/core/segment/index/column/DefaultNullValueColumnProvider.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.segment.index.column;
+
+import com.google.common.base.Preconditions;
+import org.apache.pinot.core.io.reader.impl.ConstantMultiValueInvertedIndex;
+import org.apache.pinot.core.io.reader.impl.ConstantSingleValueInvertedIndex;
+import org.apache.pinot.core.segment.index.ColumnMetadata;
+import org.apache.pinot.core.segment.index.readers.Dictionary;
+import org.apache.pinot.core.segment.index.readers.SingleDoubleDictionary;
+import org.apache.pinot.core.segment.index.readers.SingleFloatDictionary;
+import org.apache.pinot.core.segment.index.readers.SingleIntDictionary;
+import org.apache.pinot.core.segment.index.readers.SingleLongDictionary;
+import org.apache.pinot.core.segment.index.readers.SingleStringDictionary;
+import org.apache.pinot.spi.data.FieldSpec;
+import org.apache.pinot.spi.data.FieldSpec.DataType;
+
+
+/**
+ * Provide the default null value.
+ */
+public class DefaultNullValueColumnProvider extends BaseColumnProvider {
+
+  Dictionary _dictionary;
+  ColumnMetadata _columnMetadata;
+
+  public DefaultNullValueColumnProvider(ColumnContext columnContext) {
+    buildDictionary(columnContext);
+    buildMetadata(columnContext);
+    buildColumnIndexContainer(columnContext);
```

Comment:
Since column context already contains the number of rows, we can call update inverted index right here

Reply:
Not sure what you mean here. This is inside constructor, right? The current idea is to cache the provider when applicable and avoid rebuilding everything

Reply:
Hmm... ok, we rebuild this column provider everytime in the offline world, but yes. maybe caching can help (but, as you note, we need to update the inverted index, and later, any other auto-generated index if need be).